syntax = "proto3";
package patch;





/*
CreateWinPatch请求
*/
message CreateWinPatchRequest {
    /*
    补丁编号
    */
    string articleId = 1 ;
    /*
    适用系统版本(如:Windows Server 2012 R2)
    */
    repeated string osVersion = 2 ;
    /*
    操作系统架构
    */
    repeated string osArchitecture = 3 ;
    /*
    是否需要重启
    */
    bool requireReboot = 4 ;
    /*
    发布时间
    */
    string releaseTime = 5 ;
    /*
    MSRC编号
    */
    string msrc = 6 ;
    /*
    补丁大小(单位为KB)
    */
    int32 size = 7 ;
    /*
    备注
    */
    string memo = 8 ;
    /*
    补丁下载链接
    */
    string url = 9 ;
}

/*
CreateWinPatch返回
*/
message CreateWinPatchResponse {
    /*
    补丁id
    */
    string instanceId = 1 ;
}

/*
CreateWinPatchApi返回
*/
message CreateWinPatchResponseWrapper {
    /*
    返回码
    */
    int32 code = 1 ;
    /*
    返回码解释
    */
    string codeExplain = 2 ;
    /*
    错误详情
    */
    string error = 3 ;
    /*
    返回数据
    */
    patch.CreateWinPatchResponse data = 4 ;
}

